### PR TITLE
fix: Adding prepareQuery middleware into DELETE

### PR DESCRIPTION
**Pull request by wesratcliff**
#201

### CODE REVIEW

**Comment by wesratcliff** [comment]
My addition test ensures that the `query` parameter is being used by `app.delete`. 

The test deletes a single, specific Customer (of three) in the collection and then checks that only the other two customers exist.

Is this single test sufficient? `prepareQuery` has good coverage otherwise so I am unsure if duplicating existing tests adds value.

File: test/integration/delete.js [hunk 1]
```diff
@@ -83,6 +87,28 @@ module.exports = function (createFn, setup, dismantle) {
...
+            })
+          },
+          json: true
+        }, function (err, res, body) {
+          assert.ok(!err)
+          assert.equal(res.statusCode, 204)
+
+          db.models.Customer.find({}, function (err, customers) {
+            assert.equal(customers.length, 2)
+            assert.ok(customers[0].name !== 'John')
```
Comment (Zertz) [inline test/integration/delete.js]: We should iterate with forEach so we don't have to refactor this code if we had Customer objects in the future, same thing for line 204

**Comment by Zertz** [comment]
Amazing, thanks! I only have one little nitpick, see the line note above :)

**Comment by wesratcliff** [comment]
No worries -- good catch. Will adjust, squash and resubmit.

**Comment by Zertz** [comment]
Don't worry about squashing just yet

**Comment by wesratcliff** [comment]
Understood!

**Comment by Zertz** [comment]
Looks good to me! @florianholzapfel

**Comment by florianholzapfel** [comment]
perfect. thank you!

**Pull request merged**